torchaudio/datasets/librilight_limited.py

```diff
@@ -21,23 +23,16 @@ def _get_fileids_paths(path, subset, _ext_audio) -> List[Tuple[str, str]]:
         {root}/{_ARCHIVE_NAME}/1h/[0-5]/[clean, other] or
         {root}/{_ARCHIVE_NAME}/9h/[clean, other]
     """
-    if subset == "10min":
-        files_paths = [
-            (os.path.join(os.path.dirname(p), "..", ".."), str(p.stem))
-            for p in Path(path).glob("1h/0/*/*/*/*" + _ext_audio)
-        ]
-    elif subset in ["1h", "10h"]:
-        files_paths = [
-            (os.path.join(os.path.dirname(p), "..", ".."), str(p.stem))
-            for p in Path(path).glob("1h/*/*/*/*/*" + _ext_audio)
+
+    def _get_rel_folder(p, path):
+        rel_path = os.path.relpath(os.path.dirname(p), path)
+        return os.path.dirname(os.path.dirname(rel_path))  # move up 2 levels
+
+    files_paths = []
+    for folder in _SUBSET_MAP[subset]:
```

Comment:
Looks like the argument check for `subset` value is removed. Can you add that back and improve the error message to include the valid choices?

Reply:
there's an argument check for subset in init in [line 61](https://github.com/pytorch/audio/pull/2653/files#diff-e87636061c799854eb56661a88dd412f58a39305253f26feb354f2168d0aff27R61) which includes the valid choices. since this is an internal function called only from the dataset, I don't think we need to check again.

Reply:
In that case, can we use the `_SUBSET_MAP` dict at the argument check so that the change to `_SUBSET_MAP` is automatically reflected to the argument check?

Something like this;

```python
# In `__init__`
if subset not in _SUBSET_MAP:
    raise ValueError(f"`subset` must be one of {_SUBSET_MAP.keys()}. Found: {subset}")
folder = _SUBSET_MAP[subset]

_get_fileids_paths(path, folder, _ext_audio)
```

This way the use of `_SUBSET_MAP` is consolidated to only inside of `__init__`, so the question about validation won't come up again.